resources/agent.py

```diff
@@ -113,38 +117,70 @@ def post(self, id):
             return {'message': 'No file part'}, 400
 
         files = request.files.getlist('file')
+        repo = request.form.get('repo')
 
         file_contents=[]
         for file in files:
-            filename = file.filename
-            if not any([filename.endswith(filetype) for filetype in [".txt", ".pdf", ".md", ".rst"]]):
+            full_path = file.filename
+            file_id = self.get_file_id(repo, full_path)
+            if not any([file_id.endswith(filetype) for filetype in [".txt", ".pdf", ".md", ".rst"]]):
                 return {'message': 'Unsupported file type uploaded'}, 400
 
             file_content = file.stream.read()
 
-            file_contents.append([filename, file_content])
+            file_contents.append([file_id, full_path, file_content])
 
         # Add filenames to the DB
-        new_filenames = agent.filenames.copy()
+        new_full_paths = agent.filenames.copy()
         for fileinfo in file_contents:
-            new_filenames.append(fileinfo[0])
-        agent.filenames = new_filenames
+            new_full_paths.append(fileinfo[0])
+        agent.filenames = new_full_paths
         db.session.commit()
 
         def generate_progress():
-            for filename, file_content in file_contents:
-                yield json.dumps({"file": filename, "step": "start"}) + "\n"
-                extracted_text = text_extractor(filename, file_content)
-                yield json.dumps({"file": filename, "step": "text_extracted"}) + "\n"
+            for _, full_path, file_content in file_contents:
+                yield json.dumps({"file": full_path, "step": "start"}) + "\n"
+                extracted_text = text_extractor(full_path, file_content)
+                yield json.dumps({"file": full_path, "step": "text_extracted"}) + "\n"
 
                 # Only generate embeddings when there is actual texts
                 if len(extracted_text) > 0:
-                    vector_interface.add_document(extracted_text, id, filename)
-                    yield json.dumps({"file": filename, "step": "embedding_created"}) + "\n"
+                    vector_interface.add_document(extracted_text, id, repo, full_path)
+                    yield json.dumps({"file": full_path, "step": "embedding_created"}) + "\n"
 
-                yield json.dumps({"file": filename, "step": "end"}) + "\n"
+                yield json.dumps({"file": full_path, "step": "end"}) + "\n"
 
         return Response(generate_progress(), mimetype='application/json')
+    
+    def delete(self, id):
+        full_path = request.json.get("full_path")
+        repo = request.json.get("repo")
+        metadata = {"repo": repo, "agent_id": id, "full_path": full_path}
+        query = text(f"SELECT id FROM langchain_pg_embedding WHERE cmetadata='{json.dumps(metadata)}';")
```

Comment:
Since we already maintain 2 databases (the vector collection and the Agents DB), then I wonder if it would be better to store the document UUID in our Agent DB at the time we create it. Then, when we go to delete it, we would not need to query this langchain table at all, right? We would already know the document UUID to delete.

I am questioning whether we should be maintaining 2 databases at all :laughing: Seems like having a single source of truth related to "what documents are stored" would be the best (what I mean is, why track documents in the Agent DB at all? Why not just use the PGVector for this). But I think if we choose to do that, it would be a refactor that we address in a follow-up PR, not this one :)

Reply:
Since documents are broken into chunks, we would need a 3rd to map those together.
The agent knows the filenames so that the frontend can display them. We could refactor it so the agent doesn't know the file names, and then the frontend could query the embedded table and get the filenames from the metadata.

Reply:
Ok let's leave all this as is for now :)